src/core/thread/mle.cpp

```diff
@@ -3127,6 +3127,9 @@ otError Mle::HandleDiscoveryResponse(const Message &aMessage, const Ip6::Message
                     ansi.Update(mfgEUI64.m8[i]);
                 }
 
+                // Verify that length is non zero, so we don't crash when matching up bloom filter below.
+                VerifyOrExit(steeringData.GetLength() != 0, error = OT_ERROR_PARSE);
```

Comment:
Move this check into `SteeringDataTlv::IsValid()`?

Reply:
Yes, @abtink and I considered that and we thought length < 0 may still be valid. I'll move it there.